integration/setup/client.go

```diff
@@ -77,10 +77,24 @@ func mongoDBURI(tb testing.TB, opts *mongoDBURIOpts) string {
 	return u.String()
 }
 
-// makeClient returns new client for the given working MongoDB URI.
-func makeClient(ctx context.Context, uri string) (*mongo.Client, error) {
-	clientOpts := options.Client().ApplyURI(uri)
+// makeClient returns new client for the given MongoDB URI and extra options.
+// The extraOpts that comes later overwrites value for the query key.
+func makeClient(ctx context.Context, uri string, extraOpts url.Values) (*mongo.Client, error) {
```

Comment:
```suggestion
func makeClient(ctx context.Context, uri string) (*mongo.Client, error) {
```

Options should already be in the `uri`